zooniverse/front-end-monorepo
# lib-classifier: Refactor drawing task to support dropdown subtask

## Package
- lib-classifier

## Linked Issue and/or Talk Post
- closes #4902 

## Describe your changes
- add dropdown-simple (includes legacy dropdown) subtask support to Tool model
- add dropdown-simple (includes legacy dropdown) subtask support to Mark model
- refactor SubTaskPopup for dropdown-simple / dropdownSimple subtask support
- update Tool and Mark tests for dropdown subtask

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
  - https://local.zooniverse.org:8080/?project=4312&env=production&demo=true&workflow=18965
- What user actions should my reviewer step through to review this PR?
  - draw rectangle
  - select an option from dropdown subtask
  - save and close
  - reopen mark to confirm selection persists
  - click done and review classification annotation
- Which storybook stories should be reviewed?
  - any drawing tool, subtask story, like http://localhost:6006/?path=/story/drawing-tools-point--subtask
- Are there plans for follow up PR’s to further fix this bug or develop this feature? no

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Bug Fix
